qa/L0_perf_analyzer_report/test.sh

```diff
@@ -73,13 +73,7 @@ function check_cache_output {
     ERROR_STRING="Cache hit count: 0"
     num_cache_hit_lines=$(cat ${CLIENT_LOG} |  grep -i "${CACHE_STRING}" | wc -l)
     num_cache_hit_zero_lines=$(cat ${CLIENT_LOG} |  grep -i "${ERROR_STRING}" | wc -l)
-    # Top-level ensemble model requests do not currently support caching and
-    # will always report a cache hit count of zero if any composing model
-    # has caching enabled. So we check that at least one model reports
-    # non-zero cache hits for now.
-    # TODO: When ensemble models support cache hits, this should just fail
-    #       for any occurrence of ERROR_STRING
-    if [ ${num_cache_hit_lines} -eq ${num_cache_hit_zero_lines} ]; then
+    if [ ${num_cache_hit_zero_lines} -eq ${num_cache_hit_lines} ]; then
```

Comment:
Based on the previous comment in the code, I think this test case was supposed to fail if **any** models had a cache hit of zero count, rather than all, after adding this feature. However, upon second thought, I suppose if ensemble model has caching enabled, then it will intercept the cache hits before the composing models get any. Just commenting to double check your opinion on the logic we should test here.

Reply:
I made changes in Perf analyzer, which will return 0 when the composing model is not executed in case of a cache hit. Keeping that in mind, I left the condition as is, as there will be cache hit: 0 in the stats but from the composing model.
Reason for the change: There's a check in Perf analyzer expecting stats for all models.
Now that I think of it, I shouldn't return 0 for composing models, when they are not executed.
I will think about this again and the best way to fix.

Reply:
It could be reasonable to return 0 for composing models if they have caching enabled but don't get executed. 

But another way to think about it from testing perspective is to have at least two test cases for ensemble caching:
1. Ensemble caching enabled and composing models caching enabled -- expect **exactly 1** non-zero cache hit line (for the ensemble), and N-1 zero cache hit lines (composing models that didn't get executed but had caching enabled)
2. Ensemble caching disabled, but composing models caching enabled -- expect exactly N or N-1 non-zero cache hit lines in the output (depending on whether we output a line for the top-level ensemble or not - would need to see some example outputs for this)

Reply:
Makes sense. Will make the changes.

Reply:
After some thinking, I think this test should be added to L0_response_cache, since this check is related to the behavior of cache after adding top-level response caching feature, but not how Perf_analyzer handles the change.
Already, the first case is handled in the L0_response_cache with the test **"test_all_models_with_cache_enabled"**
where I enable cache in all models and check if cache hit count is not greater than inference count.
I can quickly add the second scenario too.
Let me know your thoughts on this.

Reply:
Sure, feel free to augment the L0_response_cache for more of this checking. Let's leave this check as-is in L0_perf_analyzer_report.